tests/tx/test_verification.py

```diff
@@ -0,0 +1,448 @@
+            outputs=[TxOutput(value=6400, script=b'')],
+            aux_pow=BitcoinAuxPow.dummy(),
+            parents=[
+                self._settings.GENESIS_BLOCK_HASH,
+                self._settings.GENESIS_TX1_HASH,
+                self._settings.GENESIS_TX2_HASH
+            ],
+        )
+
+    def _get_valid_tx(self) -> Transaction:
+        genesis_private_key = get_genesis_key()
+        genesis_public_key = genesis_private_key.public_key()
+        genesis_block = self.manager.tx_storage.get_transaction(self._settings.GENESIS_BLOCK_HASH)
+
+        utxo = genesis_block.outputs[0]
+        address = get_address_from_public_key(genesis_public_key)
+        script = P2PKH.create_output_script(address)
+        output = TxOutput(utxo.value, script)
+        _input = TxInput(self._settings.GENESIS_BLOCK_HASH, 0, b'')
+
+        tx = Transaction(
+            hash=b'some_hash',
+            storage=self.manager.tx_storage,
+            weight=1,
+            inputs=[_input],
+            outputs=[output],
+            parents=[
+                self._settings.GENESIS_TX1_HASH,
+                self._settings.GENESIS_TX2_HASH,
+            ]
+        )
+
+        data_to_sign = tx.get_sighash_all()
+        assert self.manager.wallet
+        public_bytes, signature = self.manager.wallet.get_input_aux_data(data_to_sign, genesis_private_key)
+        _input.data = P2PKH.create_input_data(public_bytes, signature)
+
+        return tx
+
+    def _get_valid_token_creation_tx(self) -> TokenCreationTransaction:
+        add_blocks_unlock_reward(self.manager)
+        assert self.manager.wallet
+        return create_tokens(self.manager, self.manager.wallet.get_unused_address())
+
+    def test_block_verify_basic(self) -> None:
+        block = self._get_valid_block()
+
+        verify_weight_wrapped = Mock(wraps=block.verify_weight)
+        verify_reward_wrapped = Mock(wraps=block.verify_reward)
+
+        with (
+            patch.object(Block, 'verify_weight', verify_weight_wrapped),
+            patch.object(Block, 'verify_reward', verify_reward_wrapped),
+        ):
+            self.manager.verification_service.verify_basic(block)
+
+        # Block methods
+        verify_weight_wrapped.assert_called_once()
+        verify_reward_wrapped.assert_called_once()
+
+    def test_block_verify_without_storage(self) -> None:
+        block = self._get_valid_block()
+
+        verify_pow_wrapped = Mock(wraps=block.verify_pow)
+        verify_no_inputs_wrapped = Mock(wraps=block.verify_no_inputs)
+        verify_outputs_wrapped = Mock(wraps=block.verify_outputs)
+        verify_data_wrapped = Mock(wraps=block.verify_data)
+        verify_sigops_output_wrapped = Mock(wraps=block.verify_sigops_output)
+
+        with (
+            patch.object(Block, 'verify_pow', verify_pow_wrapped),
+            patch.object(Block, 'verify_no_inputs', verify_no_inputs_wrapped),
+            patch.object(Block, 'verify_outputs', verify_outputs_wrapped),
```

Comment:
It's missing a wrapped call to the `verify_number_of_outputs()` which is usually called by `verify_outputs()`.

Reply:
Thanks!